core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java

```diff
@@ -595,22 +595,33 @@ private void testOutputEqualsInput(Type type, PartitioningMode mode1, Partitioni
     {
         TestOutputBuffer outputBuffer = new TestOutputBuffer();
         PagePartitionerBuilder pagePartitionerBuilder = pagePartitioner(outputBuffer, BIGINT, type, type);
-        PagePartitioner pagePartitioner = pagePartitionerBuilder.build();
+        PagePartitioner multiPartitionPartitioner = pagePartitionerBuilder.build();
         Page input = new Page(
                 createLongSequenceBlock(0, POSITIONS_PER_PAGE), // partition block
                 createBlockForType(type, POSITIONS_PER_PAGE),
                 createBlockForType(type, POSITIONS_PER_PAGE));
 
         List<Object> expected = readChannel(Stream.of(input, input), 1, type);
 
-        mode1.partitionPage(pagePartitioner, input);
-        mode2.partitionPage(pagePartitioner, input);
+        mode1.partitionPage(multiPartitionPartitioner, input);
+        mode2.partitionPage(multiPartitionPartitioner, input);
 
-        pagePartitioner.close();
+        multiPartitionPartitioner.close();
 
-        List<Object> partitioned = readChannel(outputBuffer.getEnqueuedDeserialized(), 1, type);
-        assertThat(partitioned).containsExactlyInAnyOrderElementsOf(expected); // output of the PagePartitioner can be reordered
+        List<Object> multiPartitionedOutput = readChannel(outputBuffer.getEnqueuedDeserialized(), 1, type);
+        assertThat(multiPartitionedOutput).containsExactlyInAnyOrderElementsOf(expected); // output of the PagePartitioner can be reordered
         outputBuffer.clear();
+
+        // Test single partition output matches input
```

Comment:
Are we exercising RLE/Dictionary specific code you added here?
If your new code is covered by existing tests didn't we loose some coverage for old code paths?

Reply:
The RLE / Dictionary specific code for `appendRange` is only called on single partitions, so no coverage was lost- I went back and added single partition testing for all the RLE / Dictionary scenarios already in `PagePartitioner` in the latest revision.